### DISCUSSION TITLE
Invalid args for response field

### DISCUSSION THREAD

**Discussion by neomn** [fastapi/fastapi#10557] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import datetime
from typing import Optional

import redis

from fastapi import FastAPI, HTTPException
from starlette.requests import Request
from starlette.responses import Response

from fastapi_cache import FastAPICache
from fastapi_cache.backends.redis import RedisBackend
from fastapi_cache.decorator import cache

from pydantic import EmailStr

from aredis_om import HashModel, NotFoundError  # <- Notice, we import from aredis_om
from aredis_om import get_redis_connection

from dotenv import dotenv_values


redis_host = dotenv_values('/app/auth_api/.env')['REDIS_HOST']
redis_port = dotenv_values('/app/auth_api/.env')['REDIS_PORT']
redis_user = dotenv_values('/app/auth_api/.env')['REDIS_USER']
redis_pass = dotenv_values('/app/auth_api/.env')['REDIS_PASSWORD']


# This Redis instance is tuned for durability.
REDIS_DATA_URL = "redis://{redis_user}:{redis_pass}@{redis_host}:{redis_port}/0"

# This Redis instance is tuned for cache performance.
REDIS_CACHE_URL = "redis://{redis_user}:{redis_pass}@{redis_host}:{redis_port}/0"


class Customer(HashModel):
    first_name: str
    last_name: str


app = FastAPI()


@app.post("/customer")
async def save_customer(customer: Customer):
    # We can save the model to Redis by calling `save()`:
    return await customer.save()  # <- We use await here


@app.get("/customers")
async def list_customers(request: Request, response: Response):
    # To retrieve this customer with its primary key, we use `Customer.get()`:
    return {"customers": await Customer.all_pks()}  # <- We also use await here


@app.get("/customer/{pk}")
@cache(expire=10)
async def get_customer(pk: str, request: Request, response: Response):
    # To retrieve this customer with its primary key, we use `Customer.get()`:
    try:
        return await Customer.get(pk)  # <- And, finally, one more await!
    except NotFoundError:
        raise HTTPException(status_code=404, detail="Customer not found")


@app.on_event("startup")
async def startup():
    r = redis.asyncio.from_url(REDIS_CACHE_URL, encoding="utf8", decode_responses=True)
    FastAPICache.init(RedisBackend(r), prefix="fastapi-cache")
    Customer.Meta.database = get_redis_connection(url=REDIS_DATA_URL, decode_responses=True)
```

### Description

i'm trying to use redis_om using [redis om official repository](https://github.com/redis/redis-om-python/blob/main/docs/fastapi_integration.md) , but it seems that there is an issue with the Model from redis_om , based of the redis_om docs :
> [Every Redis OM Model is also a Pydantic model](https://github.com/redis/redis-om-python/blob/main/docs/fastapi_integration.md#every-redis-om-model-is-also-a-pydantic-model)
Every Redis OM model is also a Pydantic model, so you can define a model and then use the model class anywhere that FastAPI expects a Pydantic model.
This means a couple of things:
A Redis OM model can be used for request body validation

 but when i try to launch the server i get this error :

> @app.post("/customer")
    field = create_response_field(
            ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/fastapi/utils.py", line 101, in create_response_field
    raise fastapi.exceptions.FastAPIError(
fastapi.exceptions.FastAPIError: Invalid args for response field

it doesn't matter if i set response_model=None, i get the same error, i cloned the repo but same error occurs, any idea on how to solve this ? 

### Operating System

Linux

### Operating System Details

running app on docker

### FastAPI Version

0.100.0

### Pydantic Version

2

### Python Version

3.11

**Comment by neomn** [comment]
[this might be the reason which cause this issue ](https://stackoverflow.com/questions/76905920/models-from-redis-om-are-not-recognized-as-basemodel-in-pydantic) 

> aredis_om uses the BaseModel class from version 1 of Pydantic. This is provided as a separate class type from pydantic v2, and [is what is being used aredis_om](https://github.com/redis/redis-om-python/blob/main/aredis_om/_compat.py#L7) - so you're not checking for the correct class.

so there no way to use pydantic v2 with redis_om ?

**Reply by sir-george2500** [reply]
Can you please show more detail about what  `customer.save()` is expected to return

**Comment by neomn** [comment]
@sir-george2500 
i don't know exactly , probably it returns the stored json object , but it doesn't matter since the problem is with the Customer model which is passed to the save_customer function, this function expect to get a pydantic v2 model but the customer is a pydantic v1 model, this is what cause the issue , even if i return a simple json response or i set response_model=None i get the same error , so it is not related to what i return

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean and organized.
If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
